expression/builtin/string_test.go

```diff
@@ -122,3 +137,32 @@ func (s *testBuiltinSuite) TestRepeat(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(v, Equals, "")
 }
+
+func (s *testBuiltinSuite) TestLowerAndUpper(c *C) {
+	v, err := builtinLower([]interface{}{nil}, nil)
+	c.Assert(err, IsNil)
+	c.Assert(v, IsNil)
+
+	v, err = builtinUpper([]interface{}{nil}, nil)
+	c.Assert(err, IsNil)
+	c.Assert(v, IsNil)
+
+	tbl := []struct {
+		Input  interface{}
+		Expect string
+	}{
+		{"abc", "abc"},
+		{1, "1"},
+	}
+
+	for _, t := range tbl {
```

Comment:
lower and upper use the same tbl?
upper("abc") should be "ABC"
lower("abd") should be "abc"